pyfar/signals/signals.py

```diff
@@ -0,0 +1,568 @@
+    elif color == "white":
+        noise = white_noise(n_samples, rms, sampling_rate, seed).time
+    else:
+        raise ValueError(f"color is {color} but must be 'pink' or 'white'.")
+
+    noise = np.tile(noise, (repetitions, 1)) if frozen else \
+        noise.reshape((repetitions, n_pulse))
+
+    # fade the noise
+    if n_fade > 0:
+        n_fade = int(n_fade)
+        fade = np.sin(np.linspace(0, np.pi/2, n_fade))**2
+        noise[..., 0:n_fade] *= fade
+        noise[..., -n_fade:] *= fade[::-1]
+
+    # add the pause
+    noise = np.concatenate((noise, np.zeros((repetitions, int(n_pause)))), -1)
+
+    # reshape to single channel signal and discard final pause
+    noise = noise.reshape((1, -1))[..., :-int(n_pause)]
+
+    # save to Signal
+    frozen_str = "frozen" if frozen else ""
+    comment = (f"{frozen_str} {color} pulsed noise signal (rms = {rms}, "
+               f"{repetitions} repetitions, {n_pulse} samples pulse duration, "
+               f"{n_pause} samples pauses, and {n_fade} samples fades.")
+
+    signal = Signal(noise, sampling_rate, fft_norm="rms", comment=comment)
+
+    return signal
+
+
+def linear_sweep(n_samples, frequency_range, n_fade=90, amplitude=1,
+                 sampling_rate=44100):
+    """Generate single channel sine sweep with linearly increasing frequency.
+
+    Time domain sweep generation according to _[1]:
+
+    .. math::
+        x(t) = \\sin(2\\pi\f_{low} t + 2\\pi (f_{high}-f_{low}) / T t^2 / 2),
+
+    with T the duration in seconds and t the sampling points in seconds.
+
+    Parameters
+    ----------
+    n_samples : int
+        The length of the sweep in samples
+    frequency_range : array like
+        Frequency range of the sweep given by the lower and upper cut-off
+        frequency in Hz.
+    n_fade : int, optional
+        The length of the squared cosine fade-out in samples. This is done to
+        avoid discontinuities at the end of the sweep. The default is 90, which
+        equals approximately 2 ms at sampling rates of 44.1 and 48 kHz.
+    amplitude : double, optional
+        The amplitude of the signal. The default is 1.
+    sampling_rate : int, optional
+        The sampling rate in Hz. The default is 44100.
+
+    Returns
+    -------
+    sweep : pyfar Signal
+        The sweep as a Signal object. The Signal is in the time domain and
+        has the 'rms' FFT normalization (see pyfar.fft.normalization).
+
+    References
+    ----------
+    .. [1]  Farina, Angelo (2000): "Simultaneous measurement of impulse
+            response and distortion with a swept-sine technique." 108th AES
+            Convention, Paris: France.
+    """
+
+    signal = _time_domain_sweep(
+        n_samples, frequency_range, n_fade, amplitude, sampling_rate, "linear")
+
+    return signal
+
+
+def exponential_sweep(n_samples, frequency_range, n_fade=90, amplitude=1,
+                      sweep_rate=None, sampling_rate=44100):
```

Comment:
1. I would rename to `n_fade_out`. Helps avoiding confusion with fade-in windows.
2. Should we include a phase angle as well?